### ISSUE TITLE
One struct for multiple image types

### ISSUE THREAD

**Issue by genuinebyte** [genuinebyte/rawproc#2]
I was a fan of the whole "separate structs for different image types" thing for a while, but I think I'm over it. It's good for controlling what kind of image you can operate on with certain functions, but types are also good for that. What, they are types? 

It's already done with `RgbImage<T>`. That `<T>` is used to describe what kind of data it holds, `f32`s or `u16`s or `u8`s or whatnot. I propose that instead of `RawImage` and `RgbImage`, a lone `Image<Kind, T>` is created, where T is `f32` or `u16`, etc. Then, there'd be something like:
```rust
enum Kind {
   Sensor,
   Rgb,
   Hsl,
}
```

That would change the current exposure signature:
```rust
pub fn exposure(rimg: &mut RawImage, ev: f32)
```
to
```rust
pub fn exposure(img: &mut Image<Kind::Sensor, f32>, ev: f32)
```

**Comment by genuinebyte** [comment]
Not sure what I was thinking with having an enum by a generic, but uh, yeah, that was dumb. Instead I made it a trait, of course, with a few zero-sized structs to match `Sensor` and `Rgb` (and soon `Hsl` and possibly `Hsv`/`Hsb`)

**Comment by genuinebyte** [comment]
Closed in 13ef9ee
